unity-client/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/Texture/AssetPromise_Texture.cs

```diff
@@ -1,32 +1,180 @@
 ﻿using System;
+using UnityEngine;
+using System.Collections;
+using DCL.Helpers;
 
 namespace DCL
 {
     public class AssetPromise_Texture : AssetPromise<Asset_Texture>
     {
+        const TextureWrapMode DEFAULT_WRAP_MODE = TextureWrapMode.Clamp;
+        const FilterMode DEFAULT_FILTER_MODE = FilterMode.Bilinear;
+
+        string url;
+        string idWithTexSettings;
+        string idWithDefaultTexSettings;
+        TextureWrapMode wrapMode;
+        FilterMode filterMode;
+        Coroutine loadCoroutine;
+        bool storeDefaultTextureInAdvance = false;
+
+        public AssetPromise_Texture(string textureUrl, TextureWrapMode textureWrapMode = DEFAULT_WRAP_MODE, FilterMode textureFilterMode = DEFAULT_FILTER_MODE, bool storeDefaultTextureInAdvance = false)
+        {
+            url = textureUrl;
+            wrapMode = textureWrapMode;
+            filterMode = textureFilterMode;
+            this.storeDefaultTextureInAdvance = storeDefaultTextureInAdvance;
+
+            idWithDefaultTexSettings = ConstructId(url, DEFAULT_WRAP_MODE, DEFAULT_FILTER_MODE);
+            idWithTexSettings = UsesDefaultWrapAndFilterMode() ? idWithDefaultTexSettings : ConstructId(url, wrapMode, filterMode);
+        }
+
         protected override void OnAfterLoadOrReuse()
         {
-            throw new NotImplementedException();
         }
 
         protected override void OnBeforeLoadOrReuse()
         {
-            throw new NotImplementedException();
+        }
+
+        internal override void Load()
+        {
+            if (state == AssetPromiseState.LOADING || state == AssetPromiseState.FINISHED)
```

Comment:
This should work, however i'm wary that we are duplicating too much code here for the changes wanted. If we try to keep most of this logic in the AssetPromise class, finding and fixing bugs in the long run will be easier. 

A few ideas to avoid overriding the whole method:
- Maybe we can wrap the library.Contains in a new virtual method and override here?
- Maybe we can remove the id parameter from GetAsset() and override it here so it gets the desired id?

Reply:
Done, check it out